docs/project-file-structure.md

```diff
@@ -64,30 +91,139 @@ to organize their structure
 
 Resources, scenes, scripts etc. used exclusively within that scene should be stored in the scenes subdirectory
 
+A scene and its script should always be together within the directory i.e. no /scripts/ and /scenes/ directories should exist within
+the project somewhere.
+
 Example:
 ```
 root/
 |- src/
-    |- Theme/
+    |- theme/
     |   |- ButtonNormal.tres
     |   |- ButtonHover.tres
     |   |- ButtonDisabled.tres
-    |- Scripts/
+    |- utility/
     |   |- Math.gd
-    |- Menu/
+    |- menu/
     |   |- MenuManager.tscn
     |   |- MenuManager.gd
     |   |- MainMenu/
     |   |   |- MainMenu.tscn
     |   |   |- MainMenu.gd
     |   |   |- MainMenuFont.ttf
     |   |   |- MainMenuLabel.tres
-    |   |- SettingsMenu/
+    |   |- settings_menu/
     |   |   |- Settings.tscn
     |   |   |- Settings.gd
     |   |   |- Resolution/
     |   |   |   |- ResolutionSelector.gd
-    |   |   |- Volume/
+    |   |   |- volume/
     |   |   |   |- VolumeGrid.tscn
     |   |   |   |- VolumeGrid.gd
 ```
+
+#### Naming Convention
+
+Directory names should be snake_case.
+```
+root/src/game_session/
+```
+
+File names should be PascalCase.
+```
+root/src/game_session/Map.tscn
+```
+
+### Asset Organization
+
+Note: Asset files should not be exported with Godot in a pck file. They should be downloaded via an installer or handled in a similar
+fashion to the data files and their exclusion from the export
+
+Asset files should be organized within a relevant category from root/
+```
+root/art/
+root/music/
+root/sfx/
+```
+
+Asset files should be organized in a subdirectory that specifies it's relevancy within the application.
+```
+root/art/economy/
+root/art/military/
+root/sfx/economy/
+root/sfx/technology/
+```
+
+Asset files should be prefixed by the asset type
+```
+iconCoal.png
+modelSoldier.obj
+ostTheCrown.mp3
+videoIntro.mp4
+```
+
+The prefix can be removed if the file is within a subdirectory that replaces the asset type
+```
+./iconCoal.png -> ./icon/Coal.png
+./ostTheCrown.png -> ./ost/TheCrown.mp3
+```
+
+#### Naming Convention
+
+Directories should be snake_case.
+
+Asset files should be PascalCase for their name.
+```
+TheCrown.mp3
+Coal.png
+```
+
+Asset files with a prefix should have that prefix be lowercase.
+```
+ostTheCrown.mp3
+iconCoal.png
+```
+
+### Data File Organization
+
+All data files should be in the root/common/ directory.
+
+Data files should be organized by category.
+```
+root/common/map/
+root/common/technology/
+root/common/pop/
+```
+
+Data files that are plain text should have the ov2d file extension.
+```
+root/common/pop/soldier.ov2d
+```
```

Comment:
Because it makes it more obvious as to what they are for. ov2d explicitly indicates that this is a file that holds data for OpenVic2.

We could still use txt files within the directory for various things, and having an explicit extensions means that we can recognize the purpose of a file at a glance. 

It would also allow us to differentiate files even when they are not used within the games directory eg. downloading a single file mod allows you to not lose it within the soup that Download directories tend to turn into

Reply:
> What is the motivation behind giving plain text files a bespoke file extension?

For any text document we are already reading either our own or a Godot format, calling them txt files is dumb because that's not the whole of what they are and you can't perform mime detection on a text document, its easier to manage if we have our own file extension. Not to mention it designates that it is NOT in fact a regular text document because it isn't, just as a `.cpp`, `.cs`, `.py`, or `.json` file is not a regular text document. This argument is just as applicable to `.csv` as well.

Reply:
We are going to stick with the extensions because of the above arguments. Cleared it up with @ClarkeCode in a discord voice call